lib/ansible/plugins/inventory/generator.py

```diff
@@ -73,12 +73,12 @@
 
 import os
 
+from itertools import product
+
 from ansible import constants as C
 from ansible.errors import AnsibleParserError
-from ansible.plugins.cache import FactCache
 from ansible.plugins.inventory import BaseInventoryPlugin
-
-from itertools import product
+from ansible.vars.fact_cache import FactCache
```

Comment:
we can just remove it, it is not really in use

Reply:
Removed in https://github.com/ansible/ansible/pull/49516/commits/5beae97802cd2a14bbbbb8aa0174143170ff2888